src/backend/plugins/video/warren_video/api.py

```diff
@@ -36,59 +42,44 @@ async def views(
 ) -> VideoViews:
     """Video views."""
-    query_params = {
-        "query": {
-            "bool": {
-                "filter": [
-                    {
-                        "range": {
-                            (
-                                "result.extensions."
-                                "https://w3id.org/xapi/video/extensions/time"
-                            ): {"lte": 30}
-                        }
-                    },
-                    {
-                        "term": {
-                            "verb.display.en-US.keyword": "played",
-                        }
-                    },
-                    {
-                        "term": {
-                            "object.id.keyword": video_id,
-                        }
-                    },
-                    {"range": {"timestamp": {"gte": filters.since}}},
-                    {"range": {"timestamp": {"lte": filters.until}}},
-                ],
-            }
-        },
-        "aggs": {
-            "daily_views": {
-                "date_histogram": {
-                    "field": settings.ES_INDEX_TIMESTAMP_FIELD,
-                    "calendar_interval": "day",
-                }
-            }
-        },
-        "size": 0,
+    query = {
+        "verb": PlayedVerb().id,
+        "activity": video_id,
+        "since": filters.since.isoformat(),
+        "until": filters.until.isoformat(),
     }
-    docs = await backends.es_client.search(
-        **query_params,
-        index=settings.ES_INDEX,
-    )
-    video_views = VideoViews(
-        total=docs["hits"]["total"]["value"],
-        daily_views=[
-            VideoDayViews(day=day.format("YYYY-MM-DD"))
-            for day in arrow.Arrow.range("day", filters.since, filters.until)
-        ],
-    )
 
-    # Daily views buckets are supposed to be sorted by date range from the
-    # oldest to the newest record
-    idx = 0
-    for bucket in docs["aggregations"]["daily_views"]["buckets"]:
-        idx = video_views.daily_views.index(VideoDayViews(day=bucket["key"]), idx)
-        video_views.daily_views[idx].views = bucket["doc_count"]
-    return video_views
+    try:
+        statements = list(
+            lrs_client.read(
+                target=settings.LRS_STATEMENTS_ENDPOINT, query=LRSQuery(query=query)
+            )
+        )
+    except BackendException as error:
+        logger.error("xAPI statements query failed: %s", error)
+        return VideoViews(total=0, daily_views=[])
+
+    daily_views_by_date = {
+        day.format(settings.DATE_DAY_FORMAT): VideoDayViews(
+            day=day.format(settings.DATE_DAY_FORMAT)
+        )
+        for day in arrow.Arrow.range("day", filters.since, filters.until)
+    }
+
+    for statement in statements:
+        # Do not count video played less than the configured time
+        if (
+            statement["result"]["extensions"][RESULT_EXTENSION_TIME]
+            < settings.VIDEO_PLAYED_THRESHOLD_DURATION
```

Comment:
We want to ignore played events that occurred **after** the threshold:

```suggestion
        if (
            statement["result"]["extensions"][RESULT_EXTENSION_TIME]
            > settings.VIDEO_PLAYED_THRESHOLD_DURATION
```